test/highlight/c/regular.c

```diff
@@ -2,6 +2,16 @@
 
 #define MACRO 0
 
+int c_init() { 1 }
+// the following aren't highlight correctly,
+// is this a problem with the parser?
+#define TESTMACRO (-1)
+#define lang_init() c_init()
+#define min(X,Y) ((X) < (Y) ? (X) : (Y))
+
+static test(c_str) = lang_init();
+static Map(c_str) langs = MAP_INIT;
```

Comment:
I don't understand what these two lines are supposed to mean. Where does `MAP_INIT` come from? Where does `c_str` come from? I guess `c_str` is the name of a parameter, if so you have to explicitly name its type in C. And I guess the assignment means you want to assign the "function" `lang_init` to `test`. That won't work First `lang_init` is a macro, it's just text that will be spliced in at some point, not a value; you have to splice the call into the function body inside curly braces. Second, if you want to actually assign a function to a variable, then that variable must be of a function pointer type, which is its own can of worms.

```suggestion
static test(char *c_str) {lang_init()};
static Map(char *c_str) langs {MAP_INIT;}
```

Reply:
This was mostly made by just copying stuff from the neovim source code and trying to make the parser give me the tree I wanted. Most of it probably doesn't fully make sense. I will try to clean it up a bit more tomorrow.